apps/etterlatte-vilkaarsvurdering/src/main/kotlin/vilkaarsvurdering/VilkaarsvurderingService.kt

```diff
@@ -76,16 +96,35 @@ class VilkaarsvurderingService(
                                 grunnlagVersjon = grunnlag.metadata.versjon
                             )
                         )
+
                     else ->
                         throw IllegalArgumentException(
                             "Støtter ikke vilkårsvurdering for behandlingType=${behandling.behandlingType}"
                         )
                 }
+
             SakType.OMSTILLINGSSTOENAD ->
                 throw IllegalArgumentException("Støtter ikke vilkårsvurdering for sakType=$sakType")
         }
     }
 
+    private suspend fun sjekkOgCommitVilkaarsvurdering(
+        behandlingId: UUID,
+        accessToken: String,
+        block: suspend () -> Vilkaarsvurdering
+    ): Vilkaarsvurdering {
+        val kanVilkaarsvurdere = behandlingKlient.vilkaarsvurder(behandlingId, accessToken, false)
+
+        if (!kanVilkaarsvurdere) {
+            throw BehandlingstilstandException
+        }
+
+        val vilkaarsvurdering = block()
+        behandlingKlient.vilkaarsvurder(behandlingId, accessToken, true)
```

Comment:
Hva skjer hvis man ikke får oppdatert behandling her?

Reply:
Da blir det nok noen _inconsistencies_, kan nok bruke transaksjoner for å få forhindret det.

Reply:
tenker at jeg får sett på det utanfor denne PR:en da det er en liten større greie. Per nå ser det ut som vi har lagt opp til å enkelt kunne bruke transaksjoner i DB-lagret, men å trekke det ut å ha det i service-lagret blir litt jobb det.

Reply:
Høres ut som en ny jirasak kanskje? :D

Reply:
https://jira.adeo.no/browse/EY-1505